ci: Add support for forked repo in ci scripts

stevenhorsman opened the pull request:
Update kata-containers repo constants so they can be set as shell
variables if required.

Fixes #3648

Signed-off-by: stevenhorsman <steven@uk.ibm.com>

stevenhorsman commented:
There seems to be some inconsistencies in the name of the kata-containers repo variable including `katacontainers_repo` and `kata_repo`. I picked `kata_repo` for this change and it seemed to work, but I'm happy to modify this to a different name if preferred. Thanks!

fidencio commented:
@stevenhorsman, sorry for the delay, I'll get to review this one Tomorrow.

fidencio reviewed (approved):
lgtm, thanks!

fidencio commented:
/test

fidencio commented:
@stevenhorsman, out of curiosity, what's the problem you're ultimately trying to fix?  Running the CI "locally" before submitting a PR or having some playground for development downstream and ensure that's green before you submit the PRs?  Or something else? :-)

stevenhorsman commented:
> @stevenhorsman, out of curiosity, what's the problem you're ultimately trying to fix? Running the CI "locally" before submitting a PR or having some playground for development downstream and ensure that's green before you submit the PRs? Or something else? :-)

Hey, primarily the latter. For example I created a fix for the issue that containerd was installing the wrong version (before liufpan raised the PR to upgrade it) and wanted to try out the build process locally before I wasted the community time and resources. I initially made some scripts automating bits of the developer guide, but figured that utilising the CI test scripts would help me understand them and be a more maintainable and sustainable path.
I wouldn't rule out running the whole CI pipeline 'locally' in future, but I don't need that yet :)

fidencio commented:
/test

stevenhorsman commented:
I'm not completely sure why the go install is failing, but I've modified my logic to set the repo variables if they were unset *or null* in case that was the issue.

fidencio commented:
/test-centos

stevenhorsman commented:
@fidencio, sorry to muck you around, I was just trying to fix up the commit message after I squashed the new commit as I think it broke the patch format rules with the double entry.

fidencio commented:
> @fidencio, sorry to muck you around, I was just trying to fix up the commit message after I squashed the new commit as I think it broke the patch format rules with the double entry.

That's fine, let me try to run the centos CI and see whether it'll pass.

fidencio commented:
/test-centos

stevenhorsman commented:
I re-looked at [`jenkins_job_build.sh`](https://github.com/kata-containers/tests/blob/1829610f8b11603d04a170b6736e002cd3f9d111/.ci/jenkins_job_build.sh#L95) and saw that `katacontainers_repo` and `kata_repo` weren't being used for similar things as I thought, but that in the CI run `kata_repo` was passed in as `github.com/kata-containers/tests` matching the `tests_repo`. I've changed my updates to use `katacontainers_repo` now, so hopefully that will stop the failure. Sorry for wasting your time - this is why I need an easy way to run the scripts against my fork ;)

fidencio commented:
> I re-looked at [`jenkins_job_build.sh`](https://github.com/kata-containers/tests/blob/1829610f8b11603d04a170b6736e002cd3f9d111/.ci/jenkins_job_build.sh#L95) and saw that `katacontainers_repo` and `kata_repo` weren't being used for similar things as I thought, but that in the CI run `kata_repo` was passed in as `github.com/kata-containers/tests` matching the `tests_repo`. I've changed my updates to use `katacontainers_repo` now, so hopefully that will stop the failure. Sorry for wasting your time - this is why I need an easy way to run the scripts against my fork ;)

@stevenhorsman, you're not wasting our time, come on.  We're all learning together here, and I must admit that a lot of parts of the project I've learned (and I am still learning) from reviewing PRs.  This one is not an exception to that. :-)

stevenhorsman commented:
I looks like that fix work (on Centos at least). Thanks for your patience :)

fidencio commented:
/test

fidencio commented:
> I looks like that fix work (on Centos at least). Thanks for your patience :)

Thanks for you patience, Steve! :-)
I'm just running all the tests, hopefully they'll work as expect.

stevenhorsman commented:
I saw that #3629 has merged now, so I rebased on upstream/main. Before that just the initrd and metric test jobs were failing and from what I've stood from slack that is an existing issue and not one I've introduced?

fidencio commented:
/test

stevenhorsman commented:
@fidencio, Fabiano can you re-run the tests for this PR please, it looks like the initial set all passed and then the 5 failures all timed out, so I don't think it is related to my change. Thanks!

fidencio commented:
/test

stevenhorsman commented:
[kata-containers-2.0-tests-ubuntu20.04-PR/227](http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu20.04-PR/227/console) looks like it failed on `The hypervisor stability test as part of 'Running stability test with sandbox_cgroup_only` With the error: `ERROR: hypervisor count:2 expected:1` which worked against that branch [yesterday](http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu20.04-PR/224/consoleFull).
I think it is related to #3654, rather than being an issue in my code, so not sure if I just got lucky yesterday, or unlucky today

fidencio commented:
/retest-ubuntu

fidencio commented:
/retest-firecracker

jodh-intel reviewed (approved):
Thanks @stevenhorsman.

lgtm

fidencio commented:
Howdy!

I'd like to ask you to rebase your PR atop of current main, as #3668 has been merged and it has a fix that unblocks the CI.

Please, do the rebase and feel free to `@fidencio` in case you do not have the powers to run `/test`.

Sorry for the inconvenience and happy hacking!

stevenhorsman commented:
Hey @fidencio - I've done the rebase, so when you get a chance to run the tests, that would be great. Thanks!

fidencio commented:
/test

stevenhorsman commented:
It looks like some commits have gone into main, so I've rebased

fidencio commented:
/test
